File: backend/internal/dto/data_quality.go
package dto

// DataQualityIssue is one actionable data gap (from dq-v1).
type DataQualityIssue struct {
	Code        string   `json:"code"`
	Severity    string   `json:"severity"`
	Title       string   `json:"title"`
	Detail      string   `json:"detail"`
	CTALabel    string   `json:"cta_label,omitempty"`
	CTAURL      string   `json:"cta_url,omitempty"`
	Affects     []string `json:"affects,omitempty"`
	Count       int      `json:"count,omitempty"`
	AccountID   string   `json:"account_id,omitempty"`
	AccountName string   `json:"account_name,omitempty"`
}

// MetricGateDto is the visibility/confidence gate for one decision metric.
type MetricGateDto struct {
	Metric     string   `json:"metric"`
	Visible    bool     `json:"visible"`
	Degraded   bool     `json:"degraded"`
	Confidence string   `json:"confidence"`
	Reasons    []string `json:"reasons,omitempty"`
	Missing    []string `json:"missing,omitempty"`
}

// AccountQualityDto is per-account quality rollup.
type AccountQualityDto struct {
	AccountID   string  `json:"account_id"`
	AccountName string  `json:"account_name"`
	Type        string  `json:"type"`
	Currency    string  `json:"currency"`
	Score       int     `json:"score"`
	Freshness   string  `json:"freshness"`
	Reconciled  bool    `json:"has_recent_reconcile"`
	DaysSinceTx int     `json:"days_since_last_tx"`
	Balance     float64 `json:"balance"`
	FormattedBalance string `json:"formatted_balance,omitempty"`
}

// DataQualityResponse is the Data Quality Center payload.
type DataQualityResponse struct {
	AsOf              string  `json:"as_of"`
	FormulaVersion    string  `json:"formula_version"`
	OverallScore      int     `json:"overall_score"`
	OverallConfidence string  `json:"overall_confidence"`
	Grade             string  `json:"grade"`

	CompletenessScore int `json:"completeness_score"`
	FreshnessScore    int `json:"freshness_score"`
	HygieneScore      int `json:"hygiene_score"`

	ReconciliationRate       float64 `json:"reconciliation_rate"`
	ReconciliationConfidence float64 `json:"reconciliation_confidence"`
	UncategorizedRate        float64 `json:"uncategorized_rate"`

	Issues   []DataQualityIssue  `json:"issues"`
	Accounts []AccountQualityDto `json:"accounts"`
	Gates    []MetricGateDto     `json:"gates"`

	MissingInputs           []string `json:"missing_inputs"`
	DecisionMetricsHidden   []string `json:"decision_metrics_hidden"`
	DecisionMetricsDegraded []string `json:"decision_metrics_degraded"`

	// Compat with existing DataSufficiency consumers.
	DataSufficiency *DataSufficiency `json:"data_sufficiency,omitempty"`
	Assumptions     []string         `json:"assumptions,omitempty"`
}

File: frontend/src/services/dataQuality.ts
import api from '../utils/api';
import type { DataSufficiency } from './dashboard';

export interface DataQualityIssue {
  code: string;
  severity: 'critical' | 'warning' | 'info' | string;
  title: string;
  detail: string;
  cta_label?: string;
  cta_url?: string;
  affects?: string[];
  count?: number;
  account_id?: string;
  account_name?: string;
}

export interface MetricGate {
  metric: string;
  visible: boolean;
  degraded: boolean;
  confidence: string;
  reasons?: string[];
  missing?: string[];
}

export interface AccountQuality {
  account_id: string;
  account_name: string;
  type: string;
  currency: string;
  score: number;
  freshness: string;
  has_recent_reconcile: boolean;
  days_since_last_tx: number;
  balance: number;
  formatted_balance?: string;
}

export interface DataQualityResponse {
  as_of: string;
  formula_version: string;
  overall_score: number;
  overall_confidence: string;
  grade: string;
  completeness_score: number;
  freshness_score: number;
  hygiene_score: number;
  reconciliation_rate: number;
  reconciliation_confidence: number;
  uncategorized_rate: number;
  issues: DataQualityIssue[];
  accounts: AccountQuality[];
  gates: MetricGate[];
  missing_inputs: string[];
  decision_metrics_hidden: string[];
  decision_metrics_degraded: string[];
  data_sufficiency?: DataSufficiency;
  assumptions?: string[];
}

export const dataQualityService = {
  async get(): Promise<DataQualityResponse> {
    const res = await api.get('/data-quality');
    return res.data.data;
  },
};

export function gateFor(dq: DataQualityResponse | undefined, metric: string): MetricGate | undefined {
  return dq?.gates?.find((g) => g.metric === metric);
}

export default dataQualityService;
